home-manager.nix

```diff
@@ -94,13 +94,27 @@ in {
         readOnly = true;
         description = "The final doom-emacs package";
       };
+
+      extraPackages = mkOption {
+        default = self: [ ];
+        type = hm.types.selectorFunction;
+        defaultText = "epkgs: [ ]";
+        example = literalExpression
+          "epkgs: [ epkgs.vterm epkgs.treesit-grammars.with-all-grammars ]";
+        description = ''
+          Extra packages available to Doom Emacs.
+          To let nix handle a doom dependency '(package! ...)' we can leverage the ':built-in t' argument
+          Consider the following example for 'vterm' in the doom config packages.el:
+          (package! vterm :built-in t)
```

Comment:
Using `vterm` as an example is a little strange: Unstraightened should already be installing `epkgs.vterm` if you enable the vterm module. It'll be the pinned version, but if you want to unpin it I'd argue `(unpin! vterm)` in packages.el makes more sense than `:built-in t` + `extraPackages`.

Your `treesit-grammars.with-all-grammars` example makes sense because you can't express that using `(package!)`, but since Doom+Unstraightened aren't trying to install that one you don't need the `:built-in t` either.

You'd really only need `:built-in t` to stop Unstraightened from installing something it would normally install, if you want to further customize it in a way that cannot be expressed in `packages.el`. I can't think of great examples off the top of my head, but applying patches to an ELisp package would be one. Can you tweak the wording a bit to make it more clear `:built-in t` is only needed when replacing packages, not adding extra ones?

I'd also drop `epkgs.vterm` from the example as I think it's less realistic.

Reply:
I've updated the description to be more precise. See 
https://github.com/marienz/nix-doom-emacs-unstraightened/pull/12/commits/cdd7af00b80d90b7c0c93389f0af424af2452948